contracts/xsgd_contract.scilla

```diff
@@ -225,33 +224,47 @@ transition configureMinter(minter : ByStr20, minterAllowedAmount : Uint128, init
     or_res = let tmp = negb isMasterMinter in orb isPaused tmp;
     match or_res with
     | False =>
-      minters[minter] := one;
-      minterAllowed[minter] := minterAllowedAmount;
-      e = { _eventname : "MinterConfigured"; minter : minter; amount : minterAllowedAmount; sender : initiator};
+      get_mint_allowance <- mintAllowed[minter];
+      mint_allowance =
+        match get_minter_allowance with
+        | Some allowance => allowance
+        | None => zero
+        end;
+      new_mint_allowance = builtin add mint_allowance amount;
+      minterAllowed[minter] := new_mint_allowance;
+      e = { _eventname : "IncreasedMinterAllowance"; minter : minter; new_mint_allowance : new_mint_allowance; sender : initiator};
       event e
     | True =>
-      e = { _eventname : "configureMinter FailedNotMasterMinterOrPaused"; minter : minter; minterAllowedAmount : minterAllowedAmount; sender : initiator};
+      e = { _eventname : "increaseMinterAllowance FailedNotMasterMinterOrPaused"; minter : minter; amount : amount; sender : initiator};
       event e
     end
   end
 end
-transition removeMinter(minter : ByStr20, initiator : ByStr20)
+transition decreaseMinterAllowance(minter : ByStr20, amount : Uint128, initiator : ByStr20)
   callerIsProxy = builtin eq _sender proxy_address;
   match callerIsProxy with
   | False =>
-    e = { _eventname : "removeMinter FailedNotProxy"; minter : minter; sender : initiator};
+    e = { _eventname : "decreaseMinterAllowance FailedNotProxy"; minter : minter; amount : amount; sender : initiator};
     event e
   | True  =>
+    isPaused <- paused;
     currentMasterMinter <- masterMinter;
     isMasterMinter = builtin eq initiator currentMasterMinter;
-    match isMasterMinter with
-    | True =>
-      delete minters[minter];
-      minterAllowed[minter] := zero;
-      e = { _eventname : "MinterRemoved"; minter : minter; sender : initiator};
-      event e
+    or_res = let tmp = negb isMasterMinter in orb isPaused tmp;
+    match or_res with
     | False =>
-      e = { _eventname : "removeMinter FailedNotMasterMinter"; minter : minter; sender : initiator};
+      get_mint_allowance <- mintAllowed[minter];
+      mint_allowance =
+        match get_minter_allowance with
+        | Some allowance => allowance
+        | None => zero
+        end;
+      new_mint_allowance = builtin sub mint_allowance amount;
```

Comment:
Underflow problem here.

Reply:
Have noted from @AmritKumar that Scilla's integer arithmetic implement Safe Math, that would result in a revert of transaction when set below 0

At the same time @jjcnn's comment on https://github.com/Xfers/XSGD-scilla/pull/10 that it should likely end as 0 if we do 10 - 30, effectively removing the address as a minter is correct so will make the change for their behaviour to be consistent

Reply:
We need to delete this line. As it duplicates the logic that follows.